crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -338,87 +338,89 @@ impl TryFrom<Transaction> for TxEnvelope {
-    type Signature = Signature;
-
-    fn tx_hash(&self) -> B256 {
-        self.hash
+impl alloy_consensus::Transaction for Transaction {
+    fn chain_id(&self) -> Option<ChainId> {
+        self.chain_id
     }
 
     fn nonce(&self) -> u64 {
         self.nonce
     }
 
-    fn block_hash(&self) -> Option<BlockHash> {
-        self.block_hash
+    fn gas_limit(&self) -> u64 {
+        self.gas
     }
 
-    fn block_number(&self) -> Option<u64> {
-        self.block_number
+    fn max_fee_per_gas(&self) -> u128 {
+        self.max_fee_per_gas.unwrap_or(self.gas_price.unwrap_or_default())
     }
 
-    fn transaction_index(&self) -> Option<u64> {
-        self.transaction_index
+    fn max_priority_fee_per_gas(&self) -> u128 {
+        self.max_fee_per_gas.unwrap_or(self.gas_price.unwrap_or_default())
     }
 
-    fn from(&self) -> Address {
-        self.from
+    fn max_fee_per_blob_gas(&self) -> Option<u128> {
+        self.max_fee_per_blob_gas
     }
 
-    fn to(&self) -> Option<Address> {
-        self.to
+    fn to(&self) -> TxKind {
+        self.to.into()
     }
 
     fn value(&self) -> U256 {
         self.value
     }
 
-    fn gas_price(&self) -> Option<u128> {
-        self.gas_price
+    fn input(&self) -> &[u8] {
+        &self.input
     }
 
-    fn gas(&self) -> u64 {
-        self.gas
+    fn ty(&self) -> u8 {
+        self.transaction_type.unwrap_or_default()
     }
 
-    fn max_fee_per_gas(&self) -> Option<u128> {
-        self.max_fee_per_gas
+    fn access_list(&self) -> Option<&AccessList> {
+        self.access_list.as_ref()
     }
 
-    fn max_priority_fee_per_gas(&self) -> Option<u128> {
-        self.max_priority_fee_per_gas
+    fn blob_versioned_hashes(&self) -> Option<&[B256]> {
+        self.blob_versioned_hashes.as_deref()
     }
 
-    fn max_fee_per_blob_gas(&self) -> Option<u128> {
-        self.max_fee_per_blob_gas
+    fn authorization_list(&self) -> Option<&[SignedAuthorization]> {
+        self.authorization_list.as_deref()
     }
+}
 
-    fn input(&self) -> &Bytes {
-        &self.input
+impl TransactionResponse for Transaction {
+    type Signature = Signature;
+
+    fn tx_hash(&self) -> B256 {
+        self.hash
     }
 
-    fn signature(&self) -> Option<Signature> {
-        self.signature
+    fn block_hash(&self) -> Option<BlockHash> {
+        self.block_hash
     }
 
-    fn chain_id(&self) -> Option<ChainId> {
-        self.chain_id
+    fn block_number(&self) -> Option<u64> {
+        self.block_number
     }
 
-    fn blob_versioned_hashes(&self) -> Option<Vec<B256>> {
-        self.blob_versioned_hashes.clone()
+    fn transaction_index(&self) -> Option<u64> {
+        self.transaction_index
     }
 
-    fn access_list(&self) -> Option<AccessList> {
-        self.access_list.clone()
+    fn from(&self) -> Address {
+        self.from
     }
 
-    fn transaction_type(&self) -> Option<u8> {
-        self.transaction_type
+    fn to(&self) -> Option<Address> {
```

Comment:
I think this can be removed? given that we already have `to` on consensus `Transaction`

Reply:
consensus `Transaction` returns `TxKind`. requiring every caller of `to` in rpc context to do this conversion to `Option<Address>` isn't good.

Reply:
I think this is fine? the conversion is as simple as calling `kind.to()`

and we can probably change `to` type to `TxKind` on rpc types as it's done in consensus serde https://github.com/alloy-rs/alloy/blob/599e57751fd986a4b3fb64935e80cc512b87a018/crates/consensus/src/transaction/legacy.rs#L48-L49

Reply:
following matze's directive, prio of this pr should be not to change the API. pls open an issue.